File: cyber/demo_py/demo01_talker.py
from cyber.python.cyber_py3 import cyber
from cyber.demo_base_proto.student_pb2 import Student
import time


if __name__ == "__main__":
    cyber.init()
    talker_node = cyber.Node("xx")
    talker = talker_node.create_writer("chatter", Student, 6)

    g = 0
    stu = Student()
    while not cyber.is_shutdown():
        time.sleep(1)
        stu.age = g
        stu.name = "wuxing"
        talker.write(stu)
        g = g + 1

    cyber.shutdown()

File: cyber/demo_cc/demo01_talker.cc
#include "cyber/cyber.h"
#include "cyber/demo_base_proto/student.pb.h"

using apollo::cyber::demo_base_proto::Student;

int main(int argc, char const *argv[])
{
    apollo::cyber::Init(argv[0]);

    auto talker_node = apollo::cyber::CreateNode("ergou");
    auto talker = talker_node->CreateWriter<Student>("chatter");

    uint64_t seq = 0;
    apollo::cyber::Rate rate(0.5);
    while(apollo::cyber::OK()){
        seq++;

        AINFO << "publish " << seq << " data";

        auto msg = std::make_shared<Student>();
        msg->set_name("wuxing");
        msg->set_age(seq);
        msg->set_height(1.72);
        msg->add_books("cpp");
        msg->add_books("python");
        msg->add_books("js");

        talker->Write(msg);

        rate.Sleep();
    }

    apollo::cyber::WaitForShutdown();
    return 0;
}
